services/api-server/tests/unit/_with_db/test_product.py

```diff
@@ -67,3 +70,43 @@ def _check_key_product_compatibility(request: httpx.Request, **kwargs):
         )
         assert response.status_code == status.HTTP_200_OK
     assert wallet_get_mock.call_count == len(keys)
+
+
+async def test_product_catalog(
+    client: httpx.AsyncClient,
+    mocked_catalog_service_api_base: respx.MockRouter,
+    create_fake_api_keys: Callable[[PositiveInt], AsyncGenerator[ApiKeyInDB, None]],
+) -> None:
+    assert client
+
+    keys: list[ApiKeyInDB] = [key async for key in create_fake_api_keys(2)]
+    assert len({key.product_name for key in keys}) == 2
+
+    def _get_service_side_effect(request: httpx.Request, **kwargs):
+        assert (
+            received_product := request.headers.get("x-simcore-products-name")
+        ) is not None
+        assert (
+            user_id := dict(
+                elm.split("=") for elm in request.url.query.decode().split("&")
+            ).get("user_id")
+        ) is not None
+        key: ApiKeyInDB | None = None
+        for key in keys:
```

Comment:
TIP: IMO this reads better
```python
if any(key.id_ == UserID(user_id) for key in keys):
     break
```

Reply:
I could not get that to work. I get a complaint that `break` is not inside a loop. Instead I now use
```python
assert (key := dict((UserID(key.id_), key) for key in keys).get(UserID(user_id))) is not None
```